### DISCUSSION TITLE
Prism WPF - Custom Dialog Window with Content

### DISCUSSION THREAD

**Discussion by sylvesp** [PrismLibrary/Prism#2776] [WPF]
Let me start with saying that I am aware of the possibility of registering Custom Dialog Windows and customize the **STYLE** of the custom dialog window.
[https://prismlibrary.com/docs/wpf/dialog-service.html](url)

Now I am facing the scenario that I have to have all my dialogs have - let's say - the same footer, 
E.g. All dialogs should have a Footer with a message saying "For help click the button on the right" and then a button that when clicked takes the user to some website.

Is it possible to create and register a Custom 'host' Dialog Window where **in addition to the Style** of the window I also customize the content? (add a footer)
All the Views of the Dialogs I want to be displayed have to be included between the Window header and my custom footer.

Or do I have to do some more complicated thing to achieve this? (maybe using regions?!)

**Comment by brianlagunas** [comment]
Create a custom dialog Window https://prismlibrary.com/docs/wpf/dialog-service.html#register-a-custom-dialog-window

**Reply by brianlagunas** [reply]
Oh, wait... I see what you mean. Right now, the Content of the window is completely replaced which prevents you from adding custom content to the Window.

**Reply by brianlagunas** [reply]
Well, it may be possible already.

Can you try explicitly implementing the Content property of the IDialogWindow interface and then in the setter, set the content property of a ContentPresenter you define in your custom window?

**Reply by sylvesp** [reply]
Thanks for the fast reply.
Sounds like something I should definitively try, will keep you updated :)

**Reply by Inictus** [reply]
To anyone curious, @brianlagunas 's suggestion worked like a charm.
